Errors when opening Notification area, after having visited timelines

Description of problem:

![bad](https://user-images.githubusercontent.com/289743/33888771-ed45f0b4-df45-11e7-82fc-ebd22d3d904e.png)

Errors in browser console:

```
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
d3.js:663 Error: <pattern> attribute width: A negative value is not valid. ("-250")
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
0
```


How reproducible:

Always


Steps to Reproduce:

1. Go to My Services
2. Visit a service with at least 1 VM
3. Click on the VM (to get to the detail screen, with timelines)
4. Go back to Dashboard
5. Open the Notifications area


Actual results:

Errors in the browser console.


Expected results:

No errors in the browser console - timelines should clean up when we leave the screen.


Additional info:

including backtraces..

```
23
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
attrConstant @ d3.js:663
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
d3_selectionPrototype.attr @ d3.js:652
e.default @ timeline.js:1
(anonymous) @ timeline.js:1
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
t @ timeline.js:1
d3_selectionPrototype.call @ d3.js:975
buildTimeline @ timeline.component.js:120
renderTimeline @ timeline.component.js:37
(anonymous) @ timeline.component.js:18
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
trigger @ jquery.js:8201
triggerHandler @ jquery.js:8275
(anonymous) @ angular-patternfly.js:11076
(anonymous) @ angular.js:20545
completeOutstandingRequest @ angular.js:6419
(anonymous) @ angular.js:6699
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
ZoneTask.invoke @ zone.js:488
timer @ zone.js:2040
setTimeout (async)
scheduleTask @ zone.js:2056
ZoneDelegate.scheduleTask @ zone.js:411
onScheduleTask @ zone.js:301
ZoneDelegate.scheduleTask @ zone.js:405
Zone.scheduleTask @ zone.js:236
Zone.scheduleMacroTask @ zone.js:259
(anonymous) @ zone.js:2072
proto.(anonymous function) @ zone.js:1366
Browser.self.defer @ angular.js:6697
timeout @ angular.js:20543
updateAccordionSizing @ angular-patternfly.js:11075
ctrl.$onChanges @ angular-patternfly.js:11125
triggerOnChangesHook @ angular.js:10780
(anonymous) @ angular.js:8709
$eval @ angular.js:18521
$apply @ angular.js:18620
flushOnChangesQueue @ angular.js:8705
$digest @ angular.js:18410
$apply @ angular.js:18628
(anonymous) @ angular.js:27456
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
d3.js:663 Error: <pattern> attribute width: A negative value is not valid. ("-250")
attrConstant @ d3.js:663
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
d3_selectionPrototype.attr @ d3.js:652
e.default @ timeline.js:1
(anonymous) @ timeline.js:1
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
t @ timeline.js:1
d3_selectionPrototype.call @ d3.js:975
buildTimeline @ timeline.component.js:120
renderTimeline @ timeline.component.js:37
(anonymous) @ timeline.component.js:18
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
trigger @ jquery.js:8201
triggerHandler @ jquery.js:8275
(anonymous) @ angular-patternfly.js:11076
(anonymous) @ angular.js:20545
completeOutstandingRequest @ angular.js:6419
(anonymous) @ angular.js:6699
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
ZoneTask.invoke @ zone.js:488
timer @ zone.js:2040
setTimeout (async)
scheduleTask @ zone.js:2056
ZoneDelegate.scheduleTask @ zone.js:411
onScheduleTask @ zone.js:301
ZoneDelegate.scheduleTask @ zone.js:405
Zone.scheduleTask @ zone.js:236
Zone.scheduleMacroTask @ zone.js:259
(anonymous) @ zone.js:2072
proto.(anonymous function) @ zone.js:1366
Browser.self.defer @ angular.js:6697
timeout @ angular.js:20543
updateAccordionSizing @ angular-patternfly.js:11075
ctrl.$onChanges @ angular-patternfly.js:11125
triggerOnChangesHook @ angular.js:10780
(anonymous) @ angular.js:8709
$eval @ angular.js:18521
$apply @ angular.js:18620
flushOnChangesQueue @ angular.js:8705
$digest @ angular.js:18410
$apply @ angular.js:18628
(anonymous) @ angular.js:27456
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
attrConstant @ d3.js:663
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
d3_selectionPrototype.attr @ d3.js:652
e.default @ timeline.js:1
(anonymous) @ timeline.js:1
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
t @ timeline.js:1
d3_selectionPrototype.call @ d3.js:975
buildTimeline @ timeline.component.js:120
renderTimeline @ timeline.component.js:37
(anonymous) @ timeline.component.js:18
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
trigger @ jquery.js:8201
triggerHandler @ jquery.js:8275
(anonymous) @ angular-patternfly.js:11076
(anonymous) @ angular.js:20545
completeOutstandingRequest @ angular.js:6419
(anonymous) @ angular.js:6699
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
ZoneTask.invoke @ zone.js:488
timer @ zone.js:2040
setTimeout (async)
scheduleTask @ zone.js:2056
ZoneDelegate.scheduleTask @ zone.js:411
onScheduleTask @ zone.js:301
ZoneDelegate.scheduleTask @ zone.js:405
Zone.scheduleTask @ zone.js:236
Zone.scheduleMacroTask @ zone.js:259
(anonymous) @ zone.js:2072
proto.(anonymous function) @ zone.js:1366
Browser.self.defer @ angular.js:6697
timeout @ angular.js:20543
updateAccordionSizing @ angular-patternfly.js:11075
ctrl.$onChanges @ angular-patternfly.js:11125
triggerOnChangesHook @ angular.js:10780
(anonymous) @ angular.js:8709
$eval @ angular.js:18521
$apply @ angular.js:18620
flushOnChangesQueue @ angular.js:8705
$digest @ angular.js:18410
$apply @ angular.js:18628
(anonymous) @ angular.js:27456
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
d3.js:663 Error: <rect> attribute width: A negative value is not valid. ("-250")
attrConstant @ d3.js:663
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
d3_selectionPrototype.attr @ d3.js:652
e.default @ timeline.js:1
e.default @ timeline.js:1
(anonymous) @ timeline.js:1
(anonymous) @ d3.js:962
d3_selection_each @ d3.js:968
d3_selectionPrototype.each @ d3.js:961
t @ timeline.js:1
d3_selectionPrototype.call @ d3.js:975
buildTimeline @ timeline.component.js:120
renderTimeline @ timeline.component.js:37
(anonymous) @ timeline.component.js:18
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
trigger @ jquery.js:8201
triggerHandler @ jquery.js:8275
(anonymous) @ angular-patternfly.js:11076
(anonymous) @ angular.js:20545
completeOutstandingRequest @ angular.js:6419
(anonymous) @ angular.js:6699
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
ZoneTask.invoke @ zone.js:488
timer @ zone.js:2040
setTimeout (async)
scheduleTask @ zone.js:2056
ZoneDelegate.scheduleTask @ zone.js:411
onScheduleTask @ zone.js:301
ZoneDelegate.scheduleTask @ zone.js:405
Zone.scheduleTask @ zone.js:236
Zone.scheduleMacroTask @ zone.js:259
(anonymous) @ zone.js:2072
proto.(anonymous function) @ zone.js:1366
Browser.self.defer @ angular.js:6697
timeout @ angular.js:20543
updateAccordionSizing @ angular-patternfly.js:11075
ctrl.$onChanges @ angular-patternfly.js:11125
triggerOnChangesHook @ angular.js:10780
(anonymous) @ angular.js:8709
$eval @ angular.js:18521
$apply @ angular.js:18620
flushOnChangesQueue @ angular.js:8705
$digest @ angular.js:18410
$apply @ angular.js:18628
(anonymous) @ angular.js:27456
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.js:4621
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:192
ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1540
globalZoneAwareCallback @ zone.js:1566
timeline.js:1 0
```

--- a/client/app/shared/timeline/timeline.component.js
+++ b/client/app/shared/timeline/timeline.component.js
@@ -14,9 +14,15 @@ function TimelineController ($element, $window) {
   const vm = this
   const d3 = $window.d3
 
-  angular.element($window).on('resize', function () {
+  const resizeHandler = function () {
     renderTimeline()
-  })
+  }
+
+  angular.element($window).on('resize', resizeHandler)
+
+  vm.$onDestroy = function () {
+    angular.element($window).off('resize', resizeHandler)
+  }
 
   vm.$onChanges = (changes) => {
     vm.options = changes.options.currentValue || vm.options

